### ISSUE TITLE
Load error - Version 3.5.1-S7.0 – Sponge 1.12-7.0.0-BETA-280

### ISSUE THREAD

**Issue by bwl** [Valandur/Web-API#12]
This looks to be from the change from Achievements to Advancements in 1.12

```
[15:51:02 ERROR] [Sponge]: Failed to load plugin: Web-API 3.5.1-S7.0 (from mods/webapi-3.5.1-S7.0.jar)
com.google.common.util.concurrent.ExecutionError: com.google.common.util.concurrent.ExecutionError: java.lang.NoClassDefFoundError: org/spongepowered/api/event/achievement/GrantAchievementEvent$TargetPlayer
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2215) ~[minecraft_server.1.12.jar:?]
	at com.google.common.cache.LocalCache.get(LocalCache.java:4154) ~[minecraft_server.1.12.jar:?]
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4158) ~[minecraft_server.1.12.jar:?]
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5147) ~[minecraft_server.1.12.jar:?]
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5153) ~[minecraft_server.1.12.jar:?]
	at com.google.inject.internal.FailableCache.get(FailableCache.java:48) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at com.google.inject.internal.ConstructorInjectorStore.get(ConstructorInjectorStore.java:50) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at com.google.inject.internal.ConstructorBindingImpl.initialize(ConstructorBindingImpl.java:138) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at com.google.inject.internal.InjectorImpl.initializeBinding(InjectorImpl.java:536) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at com.google.inject.internal.AbstractBindingProcessor$Processor$1.run(AbstractBindingProcessor.java:160) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at com.google.inject.internal.ProcessedBindingData.initializeBindings(ProcessedBindingData.java:44) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:123) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at com.google.inject.internal.InjectorImpl.createChildInjector(InjectorImpl.java:232) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at com.google.inject.internal.InjectorImpl.createChildInjector(InjectorImpl.java:236) ~[spongevanilla-1.12-7.0.0-BETA-280.jar:1.12-7.0.0-BETA-280]
	at org.spongepowered.server.plugin.VanillaPluginContainer.<init>(VanillaPluginContainer.java:43) ~[VanillaPluginContainer.class:1.12-7.0.0-BETA-280]
	at org.spongepowered.server.plugin.VanillaPluginManager.loadPlugin(VanillaPluginManager.java:159) ~[VanillaPluginManager.class:1.12-7.0.0-BETA-280]
	at java.util.ArrayList.forEach(ArrayList.java:1249) [?:1.8.0_131]
	at org.spongepowered.server.plugin.VanillaPluginManager.loadPlugins(VanillaPluginManager.java:86) [VanillaPluginManager.class:1.12-7.0.0-BETA-280]
	at org.spongepowered.server.SpongeVanilla.preInitialize(SpongeVanilla.java:117) [SpongeVanilla.class:1.12-7.0.0-BETA-280]
	at net.minecraft.server.dedicated.DedicatedServer.handler$onServerLoad$zno000(SourceFile:1228) [nx.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(SourceFile:114) [nx.class:?]
	at net.minecraft.server.MinecraftServer.run(SourceFile:434) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]```

**Comment by Valandur** [comment]
Thanks for reporting this. I'm currently quite busy and working on some major changes, but will address this issue with the next update.
I hope Sponge manages to implement the advancements by then.

**Comment by Valandur** [comment]
I have released a new version of the WebAPI, but sadly Sponge hasn't implemented the advancements yet. I will probably wait with releasing for SpongeAPI 7 until that is done :)

**Comment by Valandur** [comment]
Just wanted to give you a quick update. I'm currently waiting for the Advancements to be implemented: https://github.com/SpongePowered/SpongeAPI/pull/1554

**Comment by AnZaNaMa** [comment]
Just a note - I'm getting this same error message in my console on the newest version of the API 7 plugin but it's not crashing the server, just constantly spamming my console so I can't use it anymore.

**Comment by Valandur** [comment]
Are you using version 4.9.1?

**Comment by AnZaNaMa** [comment]
Yep. 4.9.1-S7.0 on SpongeForge 1.12.2-2529-7.0.0-BETA-2747

**Comment by Valandur** [comment]
The version you are using is before the implementation of enchantments. You need to use at least `SpongeForge 1.12.2-2529-7.0.0-BETA-2748`, then your error messages should be gone, and everything should work properly.

**Comment by Valandur** [comment]
This issue is resolved as of at least [Web-API v4.10.3](https://github.com/Valandur/Web-API/releases/tag/v4.10.3)
